frontend/src/metabase/common/components/EntityPicker/components/EntityPickerModal/TabsView.tsx

```diff
@@ -0,0 +1,92 @@
+import { Tabs, Icon } from "metabase/ui";
+import type { SearchResult } from "metabase-types/api";
+
+import type { EntityTab, TypeWithModel } from "../../types";
+import {
+  EntityPickerSearchTab,
+  EntityPickerSearchResults,
+} from "../EntityPickerSearch";
+
+export const TabsView = <TItem extends TypeWithModel>({
+  tabs,
+  onItemSelect,
+  searchQuery,
+  searchResults,
+  selectedItem,
+}: {
+  tabs: EntityTab[];
+  onItemSelect: (item: TItem) => void;
+  searchQuery: string;
+  searchResults: SearchResult[] | null;
+  selectedItem: TItem | null;
+}) => {
+  const hasSearchTab = !!searchQuery;
+  const defaultTab = hasSearchTab ? { model: "search" } : tabs[0];
```

Comment:
index access like `[0]` is unsafe in typescript (it should union Type | undefined, but it always returns Type). updating the tabs type like my previous comment mention partially solves this. i prefer to use something like lodash's first method, but unfortunately underscore's typings don't handle arrays with at least one length correctly...

Reply:
Comment still stands, but I won't let it block releasing this.